File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        L = list(s)
        i = 0
        j = len(L) - 1
        vowels = set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'])
        while i < j:
            if L[i] not in vowels:
                i += 1
            elif L[j] not in vowels:
                j -= 1
            else:
                L[i], L[j] = L[j], L[i]
                i += 1
                j -= 1
        return "".join(L)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.rs
use std::collections::HashSet;
impl Solution {
    pub fn reverse_vowels(s: String) -> String {
        let mut L: Vec<char> = Vec::new();
        for k in 0..s.len() as usize {
            let c: char = s.as_bytes()[k] as char;
            L.push(c);
        }
        let mut i: usize = 0;
        let mut j: usize = (L.len() as usize) - 1;
        let vowels: HashSet<char> = HashSet::from(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
        while (i < j) {
            if (!vowels.contains(&L[i])) {
                i += 1;
            }
            else if (!vowels.contains(&L[j])) {
                j -= 1;
            }
            else {
                let temp: char = L[i];
                L[i] = L[j];
                L[j] = temp;
                i += 1;
                j -= 1;
            }
        }
        let res: String = L.iter().collect();
        return res;
    }
}